apps/backend/src/reports/reports.controller.ts

```diff
@@ -58,17 +65,46 @@ export class ReportsController {
     status: HttpStatus.INTERNAL_SERVER_ERROR,
     description: 'internal server error',
   })
+  @ApiResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Some parameters were not passed correctly',
+  })
   @HttpCode(HttpStatus.OK)
   @Post()
   @ApiCookieAuth('access-token')
+  @UsePipes(
+    new ValidationPipe({
+      transform: true,
+      transformOptions: { enableImplicitConversion: true },
+    }),
+  )
   @UseGuards(AuthGuard)
   async createReport(
     @Req() req,
+    @Query() reportTarget: CreateReportQuery,
     @Body() reportData: CreateReportRequest,
   ): Promise<CreateReportResponse> {
     try {
+      if (
+        (reportData.reportType == ReportType.GUIDE && !reportTarget.guideId) ||
+        (reportData.reportType == ReportType.TRIP && !reportTarget.postId)
+      ) {
+        throw new InvalidReportFormat(
+          "Report type and reported id don't matched",
+        );
+      }
       const account: AccountMetadata = req.account;
-      return await this.reportsService.createReport(account.userId, reportData);
+      if (!(reportData.reportType == ReportType.GUIDE)) {
+        reportTarget.guideId = null;
+      }
+      if (!(reportData.reportType == ReportType.TRIP)) {
+        reportTarget.postId = null;
+      }
+      return await this.reportsService.createReport(
+        account.userId,
+        reportData,
+        reportTarget,
+      );
```

Comment:
Is this function have 2 responsibility?

Reply:
There are primary 2 ways to report:
1. OTHER and SYSTEM: these two can be reported without query params.
2. GUIDE and TRIP: these two types need query params.
But essentially, they're reported the same way, just with extra parameters.

Line 88-95 is to check if they report GUIDE or TRIP but manipulate parameters so that it went missing. In that case, it will throw error.
Line 97-102 is to nullify unnecessary fields before passing to service and so on. I'm not sure if there's a better way to do so.

Reply:
I think what line 88-102 does is parameter checking before invoking the service layer.
It looks okay,. You can move the code to a private function if you want a cleaner code, but it's up to you